src/app/api/game-sessions/route.ts

```diff
@@ -20,21 +92,297 @@ export async function GET() {
  */
 export async function POST(req: NextRequest) {
   try {
-    const json = await req.json();
+    const user = await getCurrentUser();
+    if (!user) {
+      return new Response("ERROR: Unauthorized request", { status: 401 });
+    }
+    if (user.role != "admin") {
+      return new Response("ERROR: No valid permissions", { status: 403 });
+    }
 
-    const body = insertGameSessionSchema.parse(json);
+    const body = await req.json();
+    const gameSessionDate = z.string().date().parse(body.date);
 
-    if (new Date(body.date) < new Date()) {
+    if (new Date(gameSessionDate) < new Date()) {
```

Comment:
one thing to consider when doing date comparisons is that gameSessionDate is a NZ date, while `new Date(gameSessionDate)` creates the date in a UTC context